BLSQ/iaso
# WC2-249 beneficiary types improvements

Multiple tasks here to be done on the beneficiary type list:

- replace the target of the eye button to be the list of beneficiaries 
- add a different button with a different icon to target the reference form (which was the target of the eye)
- add a permission (like forms) that allows to see the list of beneficiaries and beneficiary types but with no access to edition of the beneficiary type
- move the beneficiary type entry in the menu above the beneficiary list

I also added beneficiary type entries in the side menu to make it easier

Related JIRA tickets WC2-249

## Self proofreading checklist


## Changes

- move permissions messages in one file
- create ne permission `iaso_entity_type_write` to allow CRUD on entity types
- adapt api to allow only user with correct permission to CRUD 
- adapt front the same way
- get entity types and populate side menu with dynamic names
- change action buttons as requested
- wrote some unit test and cypress tests

## How to test

Create a user with only entities permission, check that he can only read infos
Play with the navigation between types and entities

## Print screen / video

https://github.com/BLSQ/iaso/assets/12494624/c78b23eb-8ec9-4485-ae4f-6789d2444fbc


## Print screen / video

@bmonjoie make sure this doesn't interfere with the mobile app (i don't think you can edit entity type with the app)
⚠️ to display beneficiaries by types in the menu you need to activate SHOW_BENEFICIARY_TYPES_IN_LIST_MENU on your account
